src/dbscan1d/core.py

```diff
@@ -92,14 +93,17 @@ def fit(self, X, y=None, sample_weight=None):
         """
         assert len(X.shape) == 1 or X.shape[-1] == 1, "X must be 1d array"
         assert y is None, "y parameter is ignored"
-        assert sample_weight is None, "sample weights are not yet supported"
         # get sorted array and sorted order
         ar = X.flatten()
+        if sample_weight is None:
+            sample_weight = np.ones_like(ar, dtype=np.float64)
+        else:
+            sample_weight = np.asarray(sample_weight, dtype=np.float64)
```

Comment:
I think when `sample_weight` is defined, it also needs to be sorted like the array, otherwise the weights might not be applied to the points the user intended.

I suggest replacing lines 102-13 with this:

```python
sort_inds = np.argsort(ar)
ar_sorted = ar[sort_inds]
sample_weight_sorted = sample_weight[sort_inds]
undo_sort = np.argsort(sort_inds)
```
Then passing `sample_weight_sorted` to `self_get_is_core` in line 106.

Reply:
Yes, you are right. I will fix it